### PR TITLE
Fix no such file or directory error

### PR DESCRIPTION
**Pull request by RissyRan**
# What does this PR do?

Add the logic to check if the output directory exists before opening the file. It fixes `no such file or directory` error when run [ViT model](https://github.com/huggingface/transformers/tree/8a817e1ecac6a420b1bdc701fcc33535a3b96ff5/examples/tensorflow/image-classification).

```
Traceback (most recent call last):
  File "/home/ranran/transformers/examples/tensorflow/image-classification/run_image_classification.py", line 564, in <module>
    main()
  File "/home/ranran/transformers/examples/tensorflow/image-classification/run_image_classification.py", line 546, in main
    with open(os.path.join(training_args.output_dir, "all_results.json"), "w") as f:
FileNotFoundError: [Errno 2] No such file or directory: './beans_outputs/all_results.json'
```

# How to reproduce:

```
pip install --upgrade pip
git clone https://github.com/huggingface/transformers.git
cd transformers && pip install .

pip install -r examples/tensorflow/_tests_requirements.txt
pip install -r examples/tensorflow/image-classification/requirements.txt


cd examples/tensorflow/image-classification
python3 run_image_classification.py \
    --dataset_name beans \
    --output_dir ./beans_outputs/ \
    --remove_unused_columns False \
    --do_train \
    --do_eval \
    --learning_rate 2e-5 \
    --num_train_epochs 1 \
    --per_device_train_batch_size 8 \
    --per_device_eval_batch_size 8 \
    --logging_strategy steps \
    --logging_steps 10 \
    --evaluation_strategy epoch \
    --save_strategy epoch \
    --save_total_limit 3 
```

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [X] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

@amyeroberts, @Rocketknight1

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

File: examples/tensorflow/image-classification/run_image_classification.py [hunk 1]
```diff
@@ -543,6 +543,8 @@ def compute_metrics(p):
                 logging.info(f"{metric_name}: {value:.3f}")
 
         if training_args.output_dir is not None:
+            if not os.path.exists(training_args.output_dir):
+                os.makedirs(training_args.output_dir)
```
Comment (sgugger) [inline examples/tensorflow/image-classification/run_image_classification.py:547]:
```suggestion
            os.makedirs(training_args.output_dir, exist_ok=True)
```

**Review by sgugger — APPROVED** [review]
Thanks for your PR! Left a suggestion to be consistent with how we usually do this in the rest of the lib.

**Review by Rocketknight1 — APPROVED** [review]
PR looks good to me with @sgugger's suggestion. Thank you!

File: examples/tensorflow/image-classification/run_image_classification.py (hunk 1 above)
Comment (RissyRan) [reply examples/tensorflow/image-classification/run_image_classification.py]: Sounds good to me! Updated. Thanks!

**Comment by Rocketknight1** [comment]
Looks like the CI is complaining about code style now! Can you `pip install transformers[quality]` and then `make style` in the `transformers `directory, then commit/push? That will run our code formatters and hopefully resolve the issue.

**Comment by RissyRan** [comment]
Thank you! I think the issues is resolved.

**Comment by Rocketknight1** [comment]
Yep, looks good. Thank you for the PR and the quick iteration!

**Pull request merged**